apps/aehttpserver/web/controllers/tx_controller.ex

```diff
@@ -2,42 +2,55 @@ defmodule Aehttpserver.TxController do
   use Aehttpserver.Web, :controller
   alias Aecore.Txs.Pool.Worker, as: Pool
   alias Aecore.Utils.Serialization, as: Serialization
-  alias Aecore.Chain.Worker, as: Chain
-  alias Aecore.Utils.Blockchain.BlockValidation
 
   def show(conn, params) do
     account_bin =
       params["account"]
       |> Base.decode16!()
-    user_txs = Pool.get_txs_for_address(account_bin, :no_hash)
+    user_txs = Pool.get_txs_for_address(account_bin)
+    #IO.inspect user_txs
     case user_txs do
       [] -> json(conn, [])
       _ ->
         case params["include_proof"]  do
           "true" ->
-            user_txs_and_hash = Pool.get_txs_for_address(account_bin, :add_hash)
-            merkle_tree = build_tx_tree(user_txs)
-            {size, {key, value, _}} = merkle_tree
-            proof = :gb_merkle_trees.merkle_proof(key, merkle_tree)
-            include_proof =
-            for proof <- user_txs_and_hash do
-              Map.put_new(proof, :proof, proof)
+            blocks_for_user_txs =
+            for user_tx <- user_txs do
+              Pool.get_block_by_txs_hash(user_tx.txs_hash)
             end
-            json(conn, Enum.map(include_proof, fn(tx) ->
+            merkle_trees =
+            for block <- blocks_for_user_txs do
+              build_tx_tree(block.txs)
+            end
+            user_txs_trees = Enum.zip(user_txs, merkle_trees)
+            proof =
+            for user_tx_tree <- user_txs_trees  do
+              {tx, tree} = user_tx_tree
+              {_, {key, value, _}} = tree
+              merkle_proof = :gb_merkle_trees.merkle_proof(key, tree)
+              verification =
```

Comment:
There should be the merkle path for the user to verify himself

this works for the simple example with only the coinbase transaction:
```
serialized_merkle_proof = Enum.map(Tuple.to_list(merkle_proof), fn merkle_proof_line ->
    Serialization.hex_binary(merkle_proof_line, :serialize)
end)
Map.put_new(tx, :proof, serialized_merkle_proof)
```

But it fails if there are multiple transactions in one block, here is a test case:
```
alias Aecore.Miner.Worker, as: Miner
alias Aecore.Keys.Worker, as: Keys
alias Aecore.Txs.Pool.Worker, as: Pool
Miner.resume()
Miner.suspend()
{:ok, to_account} = Keys.pubkey()
{:ok, tx} = Keys.sign_tx(to_account, 5, 1, 1)
Pool.add_transaction(tx)
Miner.resume()
Miner.suspend()
```

the problem is that `tree` is actually a tree of proves and not ` {_, {key, value, _}}` maybe you have to traverse the tree and serialize every key and value to hex